Sming/SmingCore/Digital.cpp

```diff
@@ -8,6 +8,11 @@
 #include "../SmingCore/Digital.h"
 #include "../Wiring/WiringFrameworkIncludes.h"
 
+//Store pin modes to be able
+// to detect pin writes to INPUT pins
+// and activate the pullup accordingly.
+static bool gIsPinInput[TOTAL_PINS+1] = {0};
```

Comment:
We should detect pin state dynamically, because it can be changed with different way.
This should be useful: http://esp8266.ru/esp8266-gpio-register/